cmd/cerca/cerca.go

```diff
@@ -27,24 +27,63 @@ func readAllowlist(location string) []string {
 	return processed
 }
 
-func complain(msg string) {
-	fmt.Printf("cerca: %s\n", msg)
+func inform(msg string, args ...interface{}) {
+	if len(args) > 0 {
+		fmt.Printf("%s\n", fmt.Sprintf(msg, args...))
+	} else {
+		fmt.Printf("%s\n", msg)
+	}
+}
+
+func complain(msg string, args ...interface{}) {
+	if len(args) > 0 {
+		inform(msg, args)
+	} else {
+		inform(msg)
+	}
 	os.Exit(0)
 }
 
-func main() {
+func cerca() {
 	// TODO (2022-01-10): somehow continually update veri sites by scraping merveilles webring sites || webring domain
 	var allowlistLocation string
 	var sessionKey string
 	var configPath string
 	var dataDir string
 	var dev bool
+
 	flag.BoolVar(&dev, "dev", false, "trigger development mode")
 	flag.StringVar(&allowlistLocation, "allowlist", "", "domains which can be used to read verification codes from during registration")
 	flag.StringVar(&sessionKey, "authkey", "", "session cookies authentication key")
 	flag.StringVar(&configPath, "config", "cerca.toml", "config and settings file containing cerca's customizations")
 	flag.StringVar(&dataDir, "data", "./data", "directory where cerca will dump its database")
+
+	help := `NAME:
+  cerca - lean forum software
+
+USAGE:
+  cerca [command] [options]
+
+COMMANDS:
+  admin     promote user to admin
+  migrate   manage database migrations
+  user      create new user
+  reset     reset user password
+
+GLOBAL OPTIONS:
+  -help
+        show help (default: false)
+
+OPTIONS:
+`
+	usage := func() {
+		fmt.Fprintf(os.Stderr, help)
+		flag.PrintDefaults()
+	}
+	flag.Usage = usage
+
```

Comment:
a new section in the readme that dumps the commands usage info you've written here / is output when running cerca --help would also be a good addition i think! when the readme was originally authored, there was pretty much just the one command lol :)

Reply:
@cblgh https://github.com/cblgh/cerca/pull/93/commits/da4f19c715eeb773a0d333d7b728cfa3e40af341 and [rendered](https://github.com/cblgh/cerca/blob/da4f19c715eeb773a0d333d7b728cfa3e40af341/README.md#usage)